### ISSUE TITLE
Pruning for Mb-MelGAN

### ISSUE THREAD

**Issue by ayush9096** [TensorSpeech/TensorFlowTTS#246]
Hi all,
Can you guys help me, how I can apply pruning to  the mb-MelGAN network. I want to do a channel based pruning for fast inference.

**Comment by dathudeptrai** [comment]
@ayush9096 https://www.tensorflow.org/model_optimization/guide/pruning/pruning_with_keras. Maybe you júst need wrap whole model by tfmot.sparsity.keras.prune_low_magnitude.

**Comment by ayush9096** [comment]
Okay... but there are some methods using which the number of channels can be reduced, keeping the same accuracy, using something like that may be useful to speed up.

**Comment by PiotrowskiD** [comment]
Has anyone worked on this? As far as I tried the problem with using tf tools for quantization aware training and pruning are the   custom residual stack layers (they are not compatible with the way tfmot works). It would be able to use the qat and pruning as they are supposed to work really great in tf right now. And after using quantization without qat there is a lot of noise introduced
